Plugins/Detray/include/Detray.hpp

```diff
@@ -0,0 +1,552 @@
+
+// detray geometry
+#include "detray/builders/detector_builder.hpp"
+#include "detray/io/frontend/payloads.hpp"
+#include "detray/io/frontend/detector_reader_config.hpp"
+#include "detray/io/frontend/implementation/json_readers.hpp"
+#include "detray/io/frontend/utils/detector_components_reader.hpp"
+
+#include "detray/io/common/geometry_reader.hpp"
+#include "detray/io/common/geometry_writer.hpp"
+
+#include "detray/io/json/json_reader.hpp"
+#include "detray/io/json/json_writer.hpp"
+#include "detray/io/json/json_io.hpp"
+#include "detray/io/frontend/writer_interface.hpp"
+
+// surface grid
+#include "Acts/Utilities/IAxis.hpp"
+#include "Acts/Detector/detail/IndexedSurfacesGenerator.hpp"
+
+// checks - prints
+#include "detray/utils/consistency_checker.hpp"
+#include "detray/io/frontend/detector_reader.hpp"
+#include "detray/navigation/volume_graph.hpp"
+
+
+// System include(s)
+#include <fstream>
+#include <optional>
+#include <initializer_list>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+#include <ios>
+
+
+namespace Acts {
+class IMaterialDecorator;
+}  // namespace Acts
+namespace ActsExamples {
+class IMaterialWriter;
+class IWriter;
+}  // namespace ActsExamples
+
+using namespace Acts;
+using namespace ActsExamples;
+using namespace detray;
+
+using detector_t = detector<default_metadata>;
+
+//
+//ACTS GEOMETRY TO DETRAY CONVERSION
+//use acts detector data to fill in all the payloads and convert these to a detray detector
+
+namespace {
+
+/// Find the position of the volume to point to
+///
+/// @param volume the volume to find
+/// @param the collection of volumes
+///
+/// @note return -1 if not found, to be interpreted by the caller
+int findVolume(
+    const Acts::Experimental::DetectorVolume* volume,
+    const std::vector<const Acts::Experimental::DetectorVolume*>& volumes) {
+  auto candidate = std::find(volumes.begin(), volumes.end(), volume);
+  if (candidate != volumes.end()) {
+    return std::distance(volumes.begin(), candidate);
+  }
+  return -1;
+}
+}  // namespace
+
+namespace detray{
+
+    /// These functions perform the conversion of the ACTS detector to the detray detector. 
+    /// The payloads are created and populated by traversing the initial ACTS detector object.
+
+    /// detray geometry writer function, debug purposes 
+    void detray_detector_print(const detector_t& det){
+
+        std::ofstream outputFile("data_try.json");
+        nlohmann::ordered_json out_json;
+        typename detector_t::name_map names{};
+        out_json["data"] = detray::io::geometry_writer::convert(det, names);
+        outputFile << out_json << std::endl;
+        return;
+    }
+    
+    /// @return the transform_payload(translation, rotation) of each surface/volume
+    /// @param Transform3 acts object, Transform3JsonConverter::Options
+    /// @brief convert the acts transform to detray transform payload
+    static io::transform_payload detray_converter_transf(
```

Comment:
For code in Acts we use `CamelCase` nomenclature, i.e. this method would be called like `detrayConverterTransform` - however, if we called the namespace here `DetrayConverter`, then converting a transform would be:

`DetrayConverter::convertTransform()`

which is pretty explicit what it does.